js/reference.js

```diff
@@ -346,21 +346,82 @@ var ERMrest = (function(module) {
          * or errors (TBD).
          */
         create: function(data) {
+            var self = this;
             try {
-                // TODO
                 //  verify: data is not null, data has non empty tuple set
+                verify(data, "'data' must be specified");
+                verify(data.length > 0, "'data' must have at least one row to create");
+
+                var defer = module._q.defer();
+
                 //  get the defaults list for the referenced relation's table
-                //  get the data
+                var defaults = getDefaults();
```

Comment:
I thought we decided that the client should decide what defaults to use? If you implement it like this, its not possible for a client to set the properties for those fields for which there are defaults. Is this consistent with our UX?